satellite/contact/endpoint.go

```diff
@@ -74,41 +88,42 @@ func (endpoint *Endpoint) CheckIn(ctx context.Context, req *pb.CheckInRequest) (
 	}
 	err = endpoint.service.overlay.UpdateCheckIn(ctx, nodeInfo)
 	if err != nil {
+		endpoint.log.Info("failed to update check in", zap.String("node address", req.Address), zap.String("node ID", nodeID.String()), zap.String("error", err.Error()))
 		return nil, rpcstatus.Error(rpcstatus.Internal, Error.Wrap(err).Error())
 	}
 
-	endpoint.log.Debug("checking in", zap.String("node addr", req.Address), zap.Bool("ping node success", pingNodeSuccess))
+	endpoint.log.Debug("checking in", zap.String("node addr", req.Address), zap.Bool("ping node success", pingNodeSuccess), zap.String("ping node err msg", pingErrorMessage))
 	return &pb.CheckInResponse{
 		PingNodeSuccess:  pingNodeSuccess,
 		PingErrorMessage: pingErrorMessage,
 	}, nil
 }
 
-func (endpoint *Endpoint) pingBack(ctx context.Context, req *pb.CheckInRequest, peerID storj.NodeID) (bool, string, error) {
+func (endpoint *Endpoint) pingBack(ctx context.Context, req *pb.CheckInRequest, peerID storj.NodeID) (_ bool, _ string, err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	pingNodeSuccess := true
+	var pingErrorMessage string
+
 	client, err := newClient(ctx, endpoint.service.dialer, req.Address, peerID)
 	if err != nil {
-		// if this is a network error, then return the error otherwise just report internal error
-		_, ok := err.(net.Error)
-		if ok {
-			return false, "", Error.New("failed to connect to %s: %v", req.Address, err)
-		}
-		endpoint.log.Info("pingBack internal error", zap.String("error", err.Error()))
-		return false, "", Error.New("couldn't connect to client at addr: %s due to internal error.", req.Address)
+		// If there is an error from trying to dial and ping the node, return that error as
+		// pingErrorMessage and not as the err. We want to use this info to update
+		// node contact info and do not want to terminate execution by returning an err
+		mon.Event("failed dial")
+		pingNodeSuccess = false
+		pingErrorMessage = fmt.Sprintf("failed to dial storage node (ID: %s) at address %s: %q", peerID, req.Address, err)
+		endpoint.log.Info("pingBack failed to dial storage node", zap.String("nodeID", peerID.String()), zap.String("node address", req.Address), zap.String("pingErrorMessage", pingErrorMessage), zap.String("error", err.Error()))
```

Comment:
don't we still need to return in this branch? client is nil, so the rest of the function won't work

Reply:
good point